### ISSUE TITLE
.NET Core Support?

### ISSUE THREAD

**Issue by nightblade9** [Thraka/SadConsole#137]
Hi,

I would like SadConsole to work with .NET Core, which I'm planning to build my next roguelike in.

Is this possible? [Word out on the street](https://github.com/Chris3606/GoRogue/issues/47#issuecomment-406036522) is that SadConsole uses MonoGame, which may support .NET core, so I'm optimistic.

Edit: Seems I didn't do my homework. MonoGame [has an open fork with this working](https://github.com/MonoGame/MonoGame/issues/5339#issuecomment-342686703), which is slated for v3.8.

I'm guessing the answer is "no, SadConsole doesn't support .NET Core today."

**Comment by Thraka** [comment]
I ported the code over to .NET core (standard 2.0) and go it running with little to no changes, using the experimental .NET core templates for monogame. But I'm just waiting on the official release from the monogame team before I move over to it.

**Comment by nightblade9** [comment]
Could I somehow get my hands on your experimental build?  I think the MonoGame team is going to take their time (if you check my referenced issue, it looks like they want to release it in 3.8, and 3.6 went out the door recently.)

Would you be okay with providing an otherwise "unsupported" NuGet package for this?

**Comment by Thraka** [comment]
I dont really want to create a NuGet package for this just yet. However, do you want the code? You can compile it and use it. It's only 1 month out of date, so not that much different.

**Comment by nightblade9** [comment]
Sure, that would be fantastic. If you can drop it onto a branch, that would be useful too.

**Comment by Thraka** [comment]
It's not part of the actual source, it was just a folder I dumped all the files into and created the project with.

[SadConsoleStandard2.zip](https://github.com/Thraka/SadConsole/files/2211126/SadConsoleStandard2.zip)

**Comment by nightblade9** [comment]
This looks good enough for me. The sample runs fine (it's all .NET Core Standard 2.0) at 60FPS.

I'm looking forward to the official release when .NET core is supported. Thanks for this.

**Comment by Thraka** [comment]
Minimal from SadConsole? I'm not sure I totally understand :)

**Comment by nightblade9** [comment]
Right, sorry, let me try to explain this better.

TL;DR: I butchered the `SadConsoleExample` template you included in your .zip and removed almost everything; [here's what's left](https://github.com/nightblade9/escape-to-earth/tree/afd25758d6282c2e0cfea54d75ec6cdddcb897af/EscapeToEarth).

Specifically, you may notice that I did things like:

- Delete the `Ansi` folder
- Delete the `CustomConsoles` folder
- Delete the `InptuHandling` and `Windows` folders
- Delete `ConsoleMetaData`, `Container`, `HeaderConsole`, `Theme` classes
- Delete `layeredsurfaces.layers` and `sad.png`

The result is a very very *very* minimal project, ideal as a template -- if I want to create a new .NET Core project in SadConsole, I can take the resulting (bare-bones) project and start with that.

While I have this thread "bookmarked" to use your attached zip for that purpose, it would be easier if, instead of repeating this process, the zip file just includes the minimum necessary to start.

Does that make sense?

**Comment by Thraka** [comment]
Ahhh yes :grin: That project has just been my demo/test project for all of the features of SadConsole :) Doing a template is one thing I've wanted to do. Thanks for the tips!

**Comment by nightblade9** [comment]
I tried this out but must be doing something wrong, I get an error about the `Screen` class not existing in `SadConsole`. Let me look at this and get back to you.

**Comment by Thraka** [comment]
There has been a lot of refactoring for 7.0. `Screen` is now `ScreenObject`. This may help: http://sadconsole.com/docs/what-are-screens-consoles-surfaces.html

**Comment by nightblade9** [comment]
Looks good to me. My roguelike is still in very, very early stages of development (I just have area-of-sight, dungeon/wall generation, and stairs down) but it works with that one change of `Screen` => `ScreenObject`.

**Comment by nightblade9** [comment]
Thanks for doing this!

**Comment by nightblade9** [comment]
I tried this again today and - still - because of the lack of MonoGame 3.8 (or maybe because I added both `SadConsole.Standard` and `SadConsole` to my project), adding MonoGame added the .NET 4.6.1 framework references to my .NET Core project and I couldn't get it to work.  I ended up going back to my template, upgrading the `SadConsole`  refernece to 7.2.0, changing `Screen` to `ScreenObject`, and I'm golden.

So, this tread is still useful as a reference 😁

**Comment by SadConsole** [comment]
You shouldn't add SadConsole to your project, only SadConsole.Standard. 

If you do a new monogame core project and add SadConsole.Standard NuGet package, it should be working just fine for a new project. You have to probably just migrate your code properly to the 7.0 changes.

**Comment by nightblade9** [comment]
I can't create a new MonoGame core project through the `dotnet` command-line. I can create an empty project, when then again, I run into the problem above about adding MonoGame adds the .NET 4.6.1 references to my project.

I'm probably doing something wrong, but I couldn't figure out where, so I resorted to this template. Also, the wiki page I read didn't make it clear if I should use `SadConsole` or `SadConsole.Standard`, so I ended up adding both.

**Comment by SadConsole** [comment]
You just need to install the unofficial templates for MonoGame Core:

```cli
dotnet new --install MonoGame.Template.CSharp

# in directory where you want to create the project
dotnet new mgdesktopgl
dotnet run
```

That will get you a blank ready to use MonoGame .NET Core game. Then you can install the nuget package for SadConsole.Standard

**Comment by nightblade9** [comment]
Is it possible to add this info somewhere on the wiki/docs? Here's the link I used to create my starter project (which didn't work out so well for me): http://sadconsole.com/docs/create-a-new-sadconsole-project.html

**Comment by SadConsole** [comment]
The docs were only written for the non-standard-core version. I'll have to update them for that. If you want to help me track that, please create an issue requesting a doc on how to get started with core.

Thanks!
